AppCenter/AppCenter/Internals/Context/AuthToken/MSAuthTokenContext.h

```diff
@@ -87,6 +87,17 @@ NS_ASSUME_NONNULL_BEGIN
  */
 - (void)checkIfTokenNeedsToBeRefreshed:(MSAuthTokenValidityInfo *)tokenValidityInfo;
 
+/**
+ * Finishes initialization process. Resets current token if nothing prevents it.
+ */
+- (void)finishInitialize;
```

Comment:
I would extract the initialization and start from this class that would make `preventResetAuthTokenAfterStart` be renamed to `resetAuthTokenNotRequired` and `finishInitialize` renamed to `resetAuthTokenIfRequired`

Reply:
Android already merged so would take it separately.